### PR TITLE
Add --with-stdcpp for ossfuzzer and mupdf use

### PR DESCRIPTION
**Pull request by ebraminio**
@behdad, not exactly an expert on autotools, is this even a correct path? Should I use AS_IF here?

### CODE REVIEW

**Comment by behdad** [comment]
Something like that. I'd phrase the help as "Do not try suppressing linkage to libstdc++".
We probably need an AM_CONDITIONAL and not try linking with C linker in hb/src/Makefile.am.

**Comment by ebraminio** [comment]
Not testes yet even but I guess now I am on right track?

**Comment by ebraminio** [comment]
Ok, I guess now everything theoretically should work.

**Comment by ebraminio** [comment]
PTAL

File: src/Makefile.am [hunk 1]
```diff
@@ -116,13 +124,17 @@ export_symbols = -export-symbols harfbuzz.def
 harfbuzz_def_dependency = harfbuzz.def
 libharfbuzz_la_LINK = $(CXXLINK) $(libharfbuzz_la_LDFLAGS)
 else
-# Use a C linker for GCC, not C++; Don't link to libstdc++
+if WITHSTDCPP
+libhb_ucdn_la_LINK = $(CXXLINK)
```
Comment (behdad) [inline src/Makefile.am]: hb_ucdn line is wrong here.

**Comment by behdad** [comment]
Thanks. Left a couple of comments. Lgtm otherwise.

**Pull request merged**